### ISSUE TITLE
Respect reduce motion setting for more transitions in our css

### ISSUE THREAD

**Issue by mjbvz** [microsoft/vscode#175990]
`workbench.reduceMotion` is a useful accessibly setting that reduces potentially distracting animations/transitions. However with #175983, I discovered this setting isn't currently respected by a number of our css transitions

Below is list of places to potentially adopt reduced motion. Please check to see if the transition should be disabled in reduced motion mode. This can be done by moving the `transition` to new css rule the uses  `.monaco-workbench:not(.reduce-motion)`:

```css
.monaco-workbench:not(.reduce-motion) my-selector {
     transition: ...; 
}
```

---

- [x] `src/vs/base/browser/ui/menu/menu.ts` @sbatten 
    1040,2: 	transition: transform 50ms ease;
    1318,5: 				transition: opacity 100ms linear;
    1325,5: 				transition: opacity 800ms linear;

- [x] `src/vs/base/browser/ui/sash/sash.css` @joaomoreno 
    110,2: 	transition: background-color 0.1s ease-out;

- [ ] `src/vs/base/browser/ui/scrollbar/media/scrollbars.css` @alexdima 
    18,2: 	transition: opacity 100ms linear;
    27,2: 	transition: opacity 800ms linear;

- [x] `src/vs/base/browser/ui/tree/media/tree.css` @joaomoreno 
    36,2: 	transition: border-color 0.1s linear;

- [ ] `src/vs/editor/browser/viewParts/minimap/minimap.css` @alexdima 
    9,2: 	transition: opacity 100ms linear;
    48,2: 	transition: opacity 0.5s;

- [x] `src/vs/editor/contrib/folding/browser/folding.css` @aeschli 
    11,2: 	transition: opacity 0.5s;

- [x] `src/vs/workbench/browser/actions/media/actions.css` @joaomoreno 
    31,2: 	transition: opacity 0.3s ease-out;

- [x] `src/vs/workbench/browser/parts/editor/media/editordroptarget.css` @mjbvz 
    22,2: 	transition: opacity 150ms ease-out;
    35,2: 	transition: opacity 150ms ease-out;
    48,2: 	transition: top 70ms ease-out, left 70ms ease-out, width 70ms ease-out, height 70ms ease-out, opacity 150ms ease-out;

- [x] `src/vs/workbench/browser/parts/statusbar/media/statusbarpart.css` @joaomoreno  
    14,2: 	transition: background-color 0.15s ease-out;

- [x] `src/vs/workbench/browser/parts/views/media/views.css` @sbatten 
    85,2: 	transition: 0.2s max-width ease-out;

- [x] `src/vs/workbench/contrib/feedback/browser/media/feedback.css` TODO
    149,2: 	transition: width 200ms ease-out;

- [x] `src/vs/workbench/contrib/notebook/browser/media/notebookCellInsertToolbar.css` @rebornix 
    22,2: 	transition: opacity 0.3s ease-in-out;

- [x] `src/vs/workbench/contrib/notebook/browser/media/notebookFolding.css` @rebornix 
    8,2: 	transition: opacity 0.1 s;

- [x] `src/vs/workbench/contrib/preferences/browser/media/settingsEditor2.css` @rzhao271 
    225,2: 	transition: opacity .3s;

- [x] `src/vs/workbench/contrib/scm/browser/media/dirtydiffDecorator.css` @lszomoru 
    17,2: 	transition: width 80ms linear, left 80ms linear, transform 80ms linear;
    28,2: 	transition: opacity 0.5s;
    38,2: 	transition: opacity 0.5s;
    49,2: 	transition: opacity 0.5s;
    59,2: 	transition: opacity 0.5s;
    85,2: 	transition: border-top-width 80ms linear, border-bottom-width 80ms linear, bottom 80ms linear, opacity 0.5s;
    94,2: 	transition: height 80ms linear;

- [x] `src/vs/workbench/contrib/terminal/browser/media/scrollbar.css` @meganrogge @Tyriar (not actioned because it's just transitioning color)
    12,2: 	transition: background-color 800ms linear;
    44,2: 	transition: opacity 100ms linear;
    50,2: 	transition: opacity 0ms linear;

- [ ] `src/vs/workbench/contrib/terminal/browser/media/terminal.css` @meganrogge @Tyriar 
    437,2: 	transition: left 70ms ease-out, right 70ms ease-out, top 70ms ease-out, bottom 70ms ease-out, opacity 150ms ease-out;

/cc @isidorn

**Comment by Tyriar** [comment]
Note to those doing this, the accessibility standard says the issue is around motion animation only https://www.w3.org/WAI/WCAG21/Understanding/animation-from-interactions.html

Which doesn't include simply changing color:

> Motion animation does not include changes of color, blurring, or opacity which do not change the perceived size, shape, or position of the element.

**Comment by mjbvz** [comment]
Thanks for the info @Tyriar

With reduced motion, my expectation was that all unnecessary animations would be disabled.  Fades can still catch my eye when they happen, so even though not mandated by the standard, I'd be in favor to reducing animations as much as possible in reduced motion mode (either that, or introduce an additional `reduced-animation` setting, but that seems too granular)

I will leave it up the area owns though to say what makes sense for the remaining animations

**Comment by Tyriar** [comment]
Terminal is done with #178616. Moving to just @lszomoru as it's the only non-color transition that we must handle.

> With reduced motion, my expectation was that all unnecessary animations would be disabled. Fades can still catch my eye when they happen, so even though not mandated by the standard, I'd be in favor to reducing animations as much as possible in reduced motion mode (either that, or introduce an additional reduced-animation setting, but that seems too granular)

The setting is for a particular disability though, the problem I have with that is we're forcing a less polished version for users that must use this setting. Perhaps there would be an additional options `'movement' | 'color'` in the setting to both give more options and to be clear about what the right thing to do here is. @meganrogge you might want to follow this up

**Comment by meganrogge** [comment]
I agree that we shouldn't change the color here as that's not what this setting applies to/describes, so it would be unexpected.
